### PR TITLE
[RLlib] Add lr-scheduling option to separate value function optimizer for APPO/IMPALA.

### PR DESCRIPTION
**Pull request by sven1977**
Add lr-scheduling option to separate value function optimizer for APPO/IMPALA.

Old API stack only (new API stack supports separate optimizers and learning rate (schedules) more naturally.
Setup via:
```
config.training(
    # Make sure we really have completely separate branches.
    model={
        ...
        "vf_share_layers": False,
    },
    _separate_vf_optimizer=True,
    # Separate vf learning rate.
    _lr_vf=0.00075,
    # Separate learning rate schedule.
    _lr_vf=[[0, 0.00075], [1000000, 0.00005]],
)
```

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by simonsays1980 — APPROVED** [review]
LGTM. Is this planned to ever be implemented in the new stack?

**Comment by sven1977** [comment]
The new API stack is much simpler to setup with arbitrary optimizers (covering different parts of the RLModule) and learning rate schedules than the old API stack (with its messy Policy APIs). So no, we will not specifically provide these options for the new stack. We should create an example on how to do this kind of thing on the new API stack (if this doesn't exist already ??). Will add to our `examples` do-over TODO list.
